third-party/py/pex/pex/link.py

```diff
@@ -73,7 +73,9 @@ def __init__(self, url):
     """
     purl = urlparse.urlparse(url)
     if purl.scheme == '':
-      purl = urlparse.urlparse(self._normalize(url))
+      purl = urlparse.urlparse(self._normalize(url), allow_fragments=False)
+    elif purl.scheme == 'file':
+      purl = urlparse.urlparse(url, allow_fragments=False)
```

Comment:
file:// worked previously - see _normalize() - it just adds a file:// prefix. This just makes sure fragments are ignored when there's no scheme or the scheme is file://